hazelcast/src/main/java/com/hazelcast/map/impl/LazyMapEntry.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.nio.serialization.SerializationService;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+/**
+ * A {@link java.util.Map.Entry Map.Entry} implementation which serializes/de-serializes key and value objects on demand.
+ * It is beneficial when you need to prevent unneeded serialization/de-serialization
+ * when creating a {@link java.util.Map.Entry Map.Entry}. Mainly targeted to supply a lazy entry to
+ * {@link com.hazelcast.map.EntryProcessor#process(Map.Entry)} method.
+ * <p/>
+ * <STRONG>Note that this implementation is not synchronized and is not thread-safe.</STRONG>
+ *
+ * @see com.hazelcast.map.impl.operation.EntryOperation#createMapEntry(Data, Object)
+ */
+public class LazyMapEntry implements Map.Entry, Serializable {
```

Comment:
Since we don't make any guarantee/promise about serializability of `Map.Entry` given to the `EntryProcessor`, I think we can just drop `Serializable` to make this class simpler.